packages/common/src/schema.ts

```diff
@@ -152,12 +152,25 @@ export const Font = z.record(
   }),
 );
 
+export const Plugin = z
+  .object({
+    ui: z.function().args(z.any()).returns(z.any()),
+    pdf: z.function().args(z.any()).returns(z.any()),
+    propPanel: z.object({
+      schema: z.unknown(),
+      widgets: z.record(z.any()).optional(),
+      defaultSchema: Schema,
+    }),
+    icon: z.string().optional(),
+  })
+  .passthrough();
+
 export const CommonOptions = z.object({ font: Font.optional() }).passthrough();
 
 const CommonProps = z.object({
   template: Template,
   options: CommonOptions.optional(),
-  plugins: z.record(z.object({ ui: z.any(), pdf: z.any(), propPanel: z.any() })).optional(),
```

Comment:
the problem with `z.any()` is that even the existence of the key is not checked, so doesn't offer any value.
I have used a more detaile definition of `Plugin` above and used `z.unknown()` for similar flexibility to `z.any()` but as least the key is checked with `parse()`